controllers/ingress/reconcile.go

```diff
@@ -141,9 +141,19 @@ func (r *ingressController) updateIngressStatus(ctx context.Context, ingress *ne
 		return fmt.Errorf("get pomerium-proxy service %s: %w", r.updateStatusFromService.String(), err)
 	}
 
-	ingress.Status.LoadBalancer = networkingv1.IngressLoadBalancerStatus{
-		Ingress: svcLoadBalancerStatusToIngress(svc.Status.LoadBalancer.Ingress),
+	ingress.Status.LoadBalancer = networkingv1.IngressLoadBalancerStatus{}
+
+	if len(svc.Status.LoadBalancer.Ingress) > 0 {
+		ingress.Status.LoadBalancer.Ingress = svcLoadBalancerStatusToIngress(svc.Status.LoadBalancer.Ingress)
+	} else if svc.Spec.Type == corev1.ServiceTypeNodePort {
+		// Assign ClusterIP for NodePort service
+		if svc.Spec.ClusterIP != "" {
+			ingress.Status.LoadBalancer.Ingress = append(ingress.Status.LoadBalancer.Ingress, networkingv1.IngressLoadBalancerIngress{
+				IP: svc.Spec.ClusterIP,
+			})
+		}
```

Comment:
1. probably this logic should live in `svcLoadBalancerStatusToIngress` 
2. the `if` condition as written would not update the status if we transition from > 0 to 0